src/javis_viewer.jl

```diff
@@ -286,3 +286,95 @@ function _pluto_viewer(video::Video, frames::Int, objects::Vector)
     arr = collect(get_javis_frame(video, objects, frame) for frame in 1:frames)
     return arr
 end
+
+"""
+    StreamConfig
+
+Holds the conguration for livestream, defaults to `nothing`
+
+#Fields
+- `livestreamto::Symbol` Livestream platform `:local` or `:twitch`  
+- `address::String` The IP address for the `:local` stream(ignored in case of `:twitch`)
+- `port::Int` The port for the `:local` stream(ignored in case of `:twitch`)
+- `twitch_key::String` TWITCH API key for your account
+"""
+struct StreamConfig
+    livestreamto::Symbol
+    address::String
+    port::Int
+    twitch_key::String
+end
+
+"""
+    setup_stream(;livestreamto, address, port, twitch_key)
+
+Sets up the livestream configuration
+"""
+function setup_stream(;livestreamto::Symbol = nothing, address::String = "0.0.0.0", port::Int = 8080, twitch_key::String = "")
+    StreamConfig(livestreamto, address, port, twitch_key)
+end
+
+"""
+    cancel_stream()
+
+Kills the livestreaming process
+"""
+function cancel_stream()
+    # kill the ffmpeg process
+    # ps aux | grep ffmpeg | grep test.gif | awk '{print $2}' | xargs kill -9 || true (hacky fix for bash errors)
+    try
+        run(
+            pipeline(`ps aux`, 
+            pipeline(`grep ffmpeg`,
+            pipeline(`grep stream_loop`, 
+            pipeline(`awk '{print $2}'`, `xargs kill -9`)))));
+    catch e
+        return "Not Streaming Anything Currently"
+    end
+    return "Livestream Cancelled!"
+end
+
+"""
+    _livestream(streamconfig, framerate, width, height, pathname)
+
+Internal method for livestreaming 
+"""
+function _livestream(streamconfig::StreamConfig, framerate::Int, width::Int, height::Int, pathname::String)
+    livestreamto = streamconfig.livestreamto
```

Comment:
should cancel a stream if one is running, right? 
Otherwise the user has to always cancel it manually.

Reply:
Yeah that was the original plan, but I realised, ffmpeg does that even more gracefully(internally) as compare to our `cancel_stream` method.

Reply:
How? :wink: